nncf/torch/pruning/filter_pruning/layers.py

```diff
@@ -51,13 +50,10 @@ def forward(self, **params):
 
             # For weights self.mask_applying_dim should be used, for bias dim=0
             dim = 0 if param_name == 'bias' else self.mask_applying_dim
-            if is_tracing_state():
-                with no_jit_trace():
-                    new_params.append(inplace_apply_filter_binary_mask(self.binary_filter_pruning_mask, param_value,
```

Comment:
@daniil-lyakhov @vshampor I've completely removed `inplace_apply_filter_binary_mask`, because it was needed for export only.
Also, I've corrected documentation for `apply_filter_binary_mask`. Please review one more time.

Reply:
Why `no_jit_trace` was used before than? Is it working without it? Other thing: should we remove `no_jit_trace` from the sparsity algo as well?

Reply:
removed `no_jit_trace` from sparsity as well

Reply:
> Originally, `no_jit_trace` excludes operations from ONNX model. But in case of binary mask, I didn't find a difference in ONNX model with and without it with ONNX opset starting from 8 up to 13. Maybe it was actual for old versions of pytorch.

@alexsu52 @vshampor @AlexKoff88 do you remember why it was used for binary mask before?
This code was since migration to Github.
https://github.com/openvinotoolkit/nncf/blame/372d740fcc234f77157d8f8f6d3e8b59463a423f/nncf/torch/pruning/filter_pruning/layers.py#L39

Reply:
e2e validation for filter pruning model passed (build 137) without `no_jit_trace`. Can't see any degradation if not using it.

Reply:
Honestly, I don't remember. How did you check that ONNX model is the same with and without `no_jit_trace` with ONNX opset starting from 8 up to 13. What torch versions did you check?

Reply:
There's `save_format` option in `export_model` function. For instance, one can specify `onnx_10` for opset 10. I used torch==1.12.1, we don't test and support lower versions of pytorch now.